modules/metastore/tests/src/Unit/DatasetApiDocsTest.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace Drupal\Tests\metastore\Unit;
+
+use Drupal\Core\Site\Settings;
+use Drupal\common\DkanApiDocsGenerator;
+use Drupal\common\Plugin\DkanApiDocsBase;
+use Drupal\common\Plugin\DkanApiDocsPluginManager;
+use Drupal\metastore\DatasetApiDocs;
+use Drupal\metastore\SchemaRetriever;
+use Drupal\metastore\Service;
+use Drupal\metastore\Storage\DataFactory;
+use Drupal\metastore\ValidMetadataFactory;
+use MockChain\Chain;
+use PHPUnit\Framework\TestCase;
+use RootedData\RootedJsonData;
+
+/**
+ * Unit tests for DatasetApiDocs.
+ */
+class DatasetApiDocsTest extends TestCase {
+
+  /**
+   *
+   */
+  public function testBuildDatasetApiDocsWithDkanApiBase() {
+
+    $dkanApiPath = '/foo/bar';
+
+    $settings = new Settings(['dkan_api_base' => $dkanApiPath]);
+
+    $generator = new DkanApiDocsGenerator(
+      $this->getManagerChain()->getMock(),
+      $settings
+    );
+
+    $service = $this->getServiceChain()->getMock();
+
+    $datasetDoc = new DatasetApiDocs($generator, $service, $settings);
+
+    $spec = $datasetDoc->getDatasetSpecific('123');
+
+    $this->assertTrue(is_array($spec['paths']));
+
+    $paths = array_keys($spec['paths']);
+    $expected_paths = [
+      $dkanApiPath . '/api/1/metastore/schemas/dataset/items/123',
+      $dkanApiPath . '/api/1/datastore/query/123/{index}',
+      $dkanApiPath . '/api/1/datastore/query/{distributionId}',
+      $dkanApiPath . '/api/1/datastore/sql'
+    ];
+    foreach ($expected_paths as $path) {
+      $this->assertContains($path, $paths);
+    }
+  }
+
+  /**
+   *
+   */
+  private function getSpec(): array {
+    return [
```

Comment:
Seems like there's a lot more here than is needed for the test? I'd think you really only need one path and no components/parameters.

Reply:
I know it seems like a lot, but I only put the minimum amount it for it not to fail. I started with just the 4 paths modified by the PR, but that didn't work.  The getDatabaseSpecific() function expect a LOT of arrays to exist and if they don't, it fails. I had to keep adding and adding over the course of 4 hours in order for it to complete. 

For example:
$fullSpec['paths']['/api/1/metastore/schemas/dataset/items/{identifier}']['get']; <- it expects the get which then requires a bunch of additional objects.

And then there's https://github.com/GetDKAN/dkan/blob/15a2ed0c6d0b4e5f7c94f92c72f99fc6f0d878b0/modules/metastore/src/DatasetApiDocs.php#L168 and https://github.com/GetDKAN/dkan/blob/15a2ed0c6d0b4e5f7c94f92c72f99fc6f0d878b0/modules/metastore/src/DatasetApiDocs.php#L145 all all the array values those expect (and all their dependencies).